README.md

=======
TF300t-AKBI v0.9 - Beta
===========

TF300t Android Kexecboot Installer Script

**THIS IS A FULL BETA!!! WIFI DOES NOT WORK UNDER LINUX.**

This installer script allows you to go through a kexecboot and
rootfs install running from an Android terminal. 

**If this is your first install, you only need to run menu item 1**

Menu items 5 and 6 are now optional. Choice 5 installs the default boot.cfg
 sample file, or can restore a backup named boot.cfg.old from the install
 directory. Choice 6 allows you to modify the boot.cfg
 from an Android text editor (Jota).

And although the scripts have some error checking, be
 sure, as always, to make sure you have a current backup... 


**Prerequisites:**

1- A working CROMi-X 5.4 w/ that10 or Zombi-Pop install
 on INTERNAL SD (no f2fs,data2fs or rom2sd)

2- A terminal emulator app (tested on Android Terminal Emulator)

3- Text editor (tested on Jota Editor - Not Jota+)

4- Script must be run from the extracted directory

5- Rootfs installer - USE ONLY ONE rootfs.tar.lzma file and
 it must be in installer directory


**Running this script:**

1- Download the latest zip here - coming soon

2- Extract to /sdcard or your favorite place

3- Place **ONE** "rootfs.tar.lzma" file in the extracted directory, example files here - http://forum.xda-developers.com/showthread.php?t=2387133

4- Open terminal and get root - su

5- cd to the directory you extracted the installer to

6- Run the script - sh TF300t-AKBI.sh (For a first time install select menu item 1)

7- Follow the prompts

Profit!

**Running the installer for the first time:**

*If you are using KitKat (CROMBi-KK), the rootfs installer will only make a 3GB image file. If you need a bigger image you will need to use one of the Aroma recovery based installers.*

1) Copy a "rootfs.tar.lzma" file (see examples below) to the directory you extracted the
installer to (ex. - /sdcard/TF300t-AKBI-v2.X.X/ ). There can only be **ONE** "rootfs.tar.lzma" file in the install directory at one time.

2) Using an Android terminal (suggest Android Terminal Emulator from the Playstore)
cd into the installer directory (ex. - cd /sdcard/TF300t-AKBI-v2.X.X/ ) and become superuser (su)

3) Run the script - (sh TF300t-AKBI.sh)

4) Choose menu item 1 reading carefully and follow prompts

5) Reboot and profit

After you reboot it will bring you directly to the kexecboot menu which will have 2 choices:
Android and your new rootfs. The kexecboot menu will time out after 10 seconds and boot to
the highest priority OS (default is Android). To choose an OS to boot, use either the volume
keys or dock arrow keys and select with the power button or dock enter key.


**Using after install - Advanced users:**

Installer menu item 4 allows you to install a boot.cfg from the installer directory. This can be any file named boot.cfg or boot.cfg.old (your backup file)

Installer menu item 5 gives the opportunity to manually edit boot.cfg using a text editor (installer pauses while you switch to a text editor)

=======

Credits:

_that@xda kernel zImage and initrd.img files (latest source code: https://github.com/that1/android_kernel_asus_tf700t.git or forked at https://github.com/workdowg/android_kernel_asus_tf700t.git) files where extracted from _that@xda's that10 and (that10cm CROMBi-KK) kernel blobs with no other modifications.

hardslog@xda kernel zImage and initrd.img files (latest source code: https://github.com/Hardslog/grimlock_kernel_asus_tf700t or forked at https://github.com/workdowg/grimlock_kernel_asus_tf700t) files where extracted from hardslog@xda's grimcm111 CROMBi-KK kernel blobs with no other modifications.

JoinTheRealms@xda, moreD_cn@xda and cogano@xda for their work on the kexec.blob. Current kexec.blob is courtesy of cogano@xda and includes a 10 sec time out to default to the PRIORITY=100 boot.cfg choice(moreD_CN@xda source code: https://github.com/cogano/kexecboot.git or forked at https://github.com/workdowg/kexecboot.git)

_sbdags@xda For his work on CROMi-X CROMBi-KK

# TF300t


changelog.txt
TF300t-AKBI v0.9 - Beta

TF300t Android Kexecboot Installer Script

v0.09 - 06/22/2015
--First release - Beta


TF300t-AKBI.sh
#!/system/bin/sh
#Android Kexecboot Installer - TF300t-AKBI v0.9 - Beta
# 06/23/2015
#by workdowg@xda
#This script must be run in the directory it was extracted to

# fail on errors
set -e

#Test for root access
perm=$(id|cut -b 5)
if [ "$perm" != "0" ]; then
   echo "This script must be run as root" 1>&2
   exit 1
fi
#cd into correct directory
workingdir=$(dirname "$0")
cd $workingdir
#Test for correct directory
if [ ! -f boot.cfg ] ; then
   echo "You are not running this script"
   echo "from the correct directory. Exiting"
   exit
fi
echo ""
echo ""
echo ""
echo "************************************"
echo "*   Android Kexecboot Installer "
echo "*       TF300t-AKBI v0.9 - Beta "
echo "************************************"
echo ""
echo ""
echo ""
echo ""
echo "1 - First time install ONLY - kexec blob, Android kexec kernel and boot.cfg"
echo "2 - Kexec Blob Install/Update"
echo "3 - Android Kexec Kernel Install/Update"
echo "4 - Rootfs/boot.cfg Installer "
echo "5 - Boot.cfg Install/Restore"
echo "6 - Boot.cfg Manual Editor"
echo "Any other key to exit"
echo ""
echo ""
echo "Remember menu items 1 and 4 require ONE rootfs.tar.lzma file to"
echo "be in the install directory."
echo ""
read n
case $n in
    1) if [ -d /data/media/linux/ ] ; then
    echo "Previous install detected, please use"
    echo "other menu options"
    sh ./TF300t-AKBI.sh
    fi
    sh ./firstinstall.sh ;;
    2) sh ./blob_installer.sh ;;
    3) sh ./kexec_kernel_installer.sh ;;
    4) sh ./rootfs_installer.sh ;;
    5) sh ./boot.cfg_installer.sh ;;
    6) sh ./boot.cfg_modifier.sh ;;
    *) exit
esac


blob_installer.sh
#!/system/bin/sh
#Android Kexecboot blob Installer - TF300t-AKBI v0.9 - Beta
# 06/23/2015
#by workdowg@xda
#This script must be run in the directory it was extracted to
# fail on errors
set -e
#Test for root access
perm=$(id|cut -b 5)
if [ "$perm" != "0" ]; then
   echo "This script must be run as root" 1>&2
   exit 1
fi
#cd into correct directory
workingdir=$(dirname "$0")
cd $workingdir
#Test for correct directory
if [ ! -f kexec.blob ] ; then
    echo "You are not running this script"
    echo "from the correct directory. Exiting"
    exit
fi
#kexecblob installer
echo ""
echo ""
echo ""
echo ""
echo "========================================================"
echo "Android Kexec blob installer by workdowg@xda"
echo "========================================================"
echo ""
echo ""
echo "Press enter to continue or ctr+c to exit"
read
echo "Installing Kexec blob"
echo "Flashing Kexec.blob to mmcblk0p4..."
dd if=kexec.blob of=/dev/block/mmcblk0p4
echo "Kexec blob flashed"
echo ""
echo ""
echo "Android Kexec blob installer - Done!"
echo ""
echo "Press enter to continue or ctr+c to exit"
read
sh ./TF300t-AKBI.sh




boot.cfg_modifier.sh
#!/system/bin/sh
#Android Kexecboot boot.cfg modifier Installer - TF300t-AKBI v0.9 - Beta
# 06/23/2015
#by workdowg@xda
#This script must be run in the directory it was extracted to
# fail on errors
set -e
#Test for root access
perm=$(id|cut -b 5)
if [ "$perm" != "0" ]; then
   echo "This script must be run as root" 1>&2
   exit 1
fi
#cd into correct directory
workingdir=$(dirname "$0")
cd $workingdir
#Test for correct directory
if [ ! -f boot.cfg ] ; then
    echo "You are not running this script"
    echo "from the correct directory. Exiting"
    exit
fi
#modify boot.cfg
echo ""
echo ""
echo ""
echo ""
echo "========================================================"
echo "Kexecboot boot.cfg modifier by workdowg@xda"
echo "========================================================"
echo ""
echo ""
#make temp directories
mkdir -p /data/media/0/kexectemp1
mkdir -p /data/media/0/kexectemp2
#mount mmcblk0p5
mount -t vfat /dev/block/mmcblk0p5 /data/media/0/kexectemp1/
cp /data/media/0/kexectemp1/multiboot/boot.cfg /data/media/0/kexectemp2/boot.tmp
#make boot.tmp writable by regular user
chown media_rw:media_rw /data/media/0/kexectemp2
chown media_rw:media_rw /data/media/0/kexectemp2/boot.tmp
echo "Now switch to a text editor"
echo "and carefully edit"
echo "/data/media/0/kexectemp2/boot.tmp"
echo "Then return here and press enter to continue"
echo ""
read
cp /data/media/0/kexectemp2/boot.tmp /data/media/0/kexectemp1/multiboot/boot.cfg
echo ""
echo ""
echo "Waiting for sync..."
sleep 10
umount /data/media/0/kexectemp1/ || echo "Warning: could not unmount /dev/block/mmcblk0p5"
rm -r /data/media/0/kexectemp1
rm -r /data/media/0/kexectemp2
echo "DONE! Please reboot and profit."
echo "========================================================"
echo "Press enter to continue or ctr+c to exit"
read
sh ./TF300t-AKBI.sh


boot.cfg_installer.sh
#!/system/bin/sh
#Android Kexecboot boot.cfg/restorer Installer - TF300t-AKBI v0.9 - Beta
# 06/23/2015
#by workdowg@xda
#This script must be run in the directory it was extracted to
# fail on errors
set -e
#Test for root access
perm=$(id|cut -b 5)
if [ "$perm" != "0" ]; then
   echo "This script must be run as root" 1>&2
   exit 1
fi
#cd into correct directory
workingdir=$(dirname "$0")
cd $workingdir
#Test for correct directory
if [ ! -f boot.cfg ] ; then
    echo "You are not running this script"
    echo "from the correct directory. Exiting"
    exit
fi
#boot.cfg installer 
echo ""
echo ""
echo ""
echo ""
echo "========================================================"
echo "Kexecboot boot.cfg installer/restorer by workdowg@xda"
echo "========================================================"
echo ""
echo ""
echo "With this script you can:"
echo ""
echo "1) Install default boot.cfg (and backup exsisting)"
echo "2) Restore boot.cfg.old (NO backup of exsisting)"
echo "3) Exit"
echo ""
read z
echo ""
case $z in
    1) mkdir -p /data/media/0/kexecbootcfg
		mount -t vfat /dev/block/mmcblk0p5 /data/media/0/kexecbootcfg/
		echo "Backing up boot.cfg..."
		if [ -f /data/media/0/kexecbootcfg/multiboot/boot.cfg ]
		then cp -f /data/media/0/kexecbootcfg/multiboot/boot.cfg boot.cfg.old
		fi
		mkdir -p /data/media/0/kexecbootcfg/multiboot/
		cp boot.cfg /data/media/0/kexecbootcfg/multiboot/
		echo ""
		echo ""
		echo "Copying default boot.cfg..."
		echo ""
		echo ""
		echo "Waiting for sync..."
		sleep 10
		umount /data/media/0/kexecbootcfg/
		rm -r /data/media/0/kexecbootcfg
		echo ""
		echo ""
		echo "Kexecboot boot.cfg installer - Done!"
		echo ""
		echo "Press enter to continue or ctr+c to exit"
		read
		sh ./TF300t-AKBI.sh ;;
	2)  ls *.old || echo "No boot.cfg.old found"
		echo ""
		ls
		echo ""
		echo "Enter a file to restore..."
		echo ""
		read oldfile
		if [ ! -f "$oldfile" ] ; then
		echo ""
		echo "File doesn't exist or entred wrong, run installer again."
		exit 1
		fi
		echo ""
		mkdir -p /data/media/0/kexecbootcfg
		mount -t vfat /dev/block/mmcblk0p5 /data/media/0/kexecbootcfg/
		cp -f "$oldfile" /data/media/0/kexecbootcfg/multiboot/boot.cfg
		echo ""
		echo ""
		echo "Restoring $oldfile ..."
		echo ""
		echo ""
		echo "Waiting for sync..."
		sleep 10
		umount /data/media/0/kexecbootcfg/
		rm -r /data/media/0/kexecbootcfg
		echo ""
		echo ""
		echo "Kexecboot boot.cfg installer - Done!"
		echo ""
		echo "Press enter to continue or ctr+c to exit"
		read
		sh ./TF300t-AKBI.sh ;;
    3) echo "Exiting..."
		echo ""
		exit 1 ;;
    *) echo "Invalid selection, run installer again"
		echo ""
		exit
esac
read



kexec_kernel_installer.sh
#!/system/bin/sh
#Android Kexecboot kernel Installer - TF300t-AKBI v0.9 - Beta
# 06/23/2015
#by workdowg@xda
#This script must be run in the directory it was extracted to 
#
# fail on errors
set -e
#Test for root access
perm=$(id|cut -b 5)
if [ "$perm" != "0" ]; then
   echo "This script must be run as root" 1>&2
   exit 1
fi
#cd into correct directory
workingdir=$(dirname "$0")
cd $workingdir
#Test for correct directory
if [ ! -d "CROMi-X_that10/" ] ; then
	echo ""
	echo ""
	echo ""
    echo "You are not running this script"
    echo "from the correct directory. Exiting"
    exit
fi
#Android kexec kernel installer 
echo ""
echo ""
echo ""
echo ""
echo "========================================================"
echo "Android Kexec kernel installer by workdowg@xda"
echo "========================================================"
echo ""
echo ""
echo "Please make a selection"
echo ""
echo ""
echo ""
echo "1  - CROMi-X that10 kernel"
echo "2  - CROMi-X stock kernel"
echo "3  - ZOMBi-POP - Omni base kernel"
echo "4  - ZOMBi-POP - Sbdags modded _that11  Omni kernel"
echo "5  - ZOMBi-POP - hardslog Omni Grimlock"
echo "6  - ZOMBi-POP - hardslog Omni Grimlock Lite"
echo "7  - ZOMBi-X-KK - Sbdags modded _that11  Omni kernel"
echo "8  - ZOMBi-X-KK - hardslog Omni Grimlock"
echo "9  - ZOMBi-X-KK - Omni base kernel"
echo "Any other key exits"
read kernel_ver
echo "Mounting /system r/w..."
mount -o remount,rw -t ext4 /dev/block/mmcblk0p1 /system || echo "/system not mounted r/w"
mkdir -p /system/boot
case $kernel_ver in
    1) cp CROMi-X_that10/* /system/boot/ ;;
    2) cp CROMi-X_Stock/* /system/boot/ ;;
    3) cp omni-zombi-pop/* /system/boot/ ;;
    4) cp bootinit115-stock-zombie-pop/* /system/boot/ ;;
    5) cp grim300511-zombi-pop/* /system/boot/ ;;
    6) cp grimzombilite5-zombi-pop/* /system/boot/ ;;	
    7) cp that11cm11-300-omni-kk/* /system/boot/ ;;
    8) cp grimomni4-omni-kk/* /system/boot/ ;;
    9) cp bootinit102-300-omni-kk/* /system/boot/ ;;	
    *) mount -o remount,ro -t ext4 /dev/block/mmcblk0p1 /system || echo "/system not mounted r/o"
    exit 1
esac
echo "Remount /system r/o..." 
mount -o remount,ro -t ext4 /dev/block/mmcblk0p1 /system || echo "/system not mounted r/o"
echo ""
echo ""
echo "Android Kexec kernel installer - Done!"
echo ""
echo "Press enter to continue or ctr+c to exit"
read
sh ./TF300t-AKBI.sh



firstinstall.sh
#!/system/bin/sh
#Android Kexecboot First time install Installer - TF300t-AKBI v0.9 - Beta
# 06/23/2015
#by workdowg@xda
#This script must be run in the directory it was extracted 

# fail on errors
set -e
#Test for root access
perm=$(id|cut -b 5)
if [ "$perm" != "0" ]; then
   echo "This script must be run as root" 1>&2
   exit 1
fi
#cd into correct directory
workingdir=$(dirname "$0")
cd $workingdir
#Test for correct directory
if [ ! -f boot.cfg ] ; then
    echo "You are not running this script"
    echo "from the correct directory. Exiting"
    exit
fi
#variables
export installer_path=$(pwd -P)
export kit=/data/media/linux
if [ ! -d $kit ]; then
	mkdir /data/media/linux; 
fi
export bin=/system/bin
export mnt=/data/local/mnt
if [ ! -d $mnt ]; then
	mkdir $mnt; 
fi
export PATH=$bin:/usr/bin:/usr/local/bin:/usr/sbin:/bin:/usr/local/sbin:/usr/games:$PATH
echo ""
echo ""
echo ""
 echo ""
echo "************************************"
echo "*   TF300t-AKBI first run Installer "
echo "************************************"
echo ""
echo "This will install everything needed for a successful"
echo "kexecboot installation. Kexec blob, Android kexec kernel and a boot.cfg."
echo "You will still be prompted several times. If this installer fails at any point,"
echo "you will need to run the main installer again starting with menu item 2."
echo ""
echo "Press enter to continue, ctrl+c to exit."
echo ""
read
#kexecboot kernel install
#Test for correct directory
if [ ! -f kexec.blob ] ; then
    echo "You are not running this script"
    echo "from the correct directory. You will now need to"
    echo "run the main main installer starting with menu item 2."
    echo "Exiting"
    echo ""
    exit
fi
echo ""
echo ""
echo ""
echo ""
echo "========================================================"
echo "Android Kexec blob installer by workdowg@xda"
echo "========================================================"
echo ""
echo ""
echo ""
echo "Installing Kexec blob"
echo "Flashing Kexec.blob to mmcblk0p4..."
dd if=kexec.blob of=/dev/block/mmcblk0p4
echo "Kexec blob flashed"
echo ""
echo ""
echo "Android Kexec blob installer - Done!"
echo ""
echo ""
#kexecboot Android kernel install
#Test for correct directory
if [ ! -d "CROMi-X_that10/" ] ; then
	echo "You are not running this script"
    echo "from the correct directory. You will need to"
    echo "run the main installer again starting with menu item 3."
    echo "Exiting"
    echo ""
    exit
fi
echo ""
echo ""
echo ""
echo ""
echo "========================================================"
echo "Android Kexec kernel installer by workdowg@xda"
echo "========================================================"
echo ""
echo ""
echo "Please make a selection"
echo ""
echo ""
echo ""
echo "1  - CROMi-X that10 kernel"
echo "2  - CROMi-X stock kernel"
echo "3  - ZOMBi-POP - Omni base kernel"
echo "4  - ZOMBi-POP - Sbdags modded _that11  Omni kernel"
echo "5  - ZOMBi-POP - hardslog Omni Grimlock"
echo "6  - ZOMBi-POP - hardslog Omni Grimlock Lite"
echo "7  - ZOMBi-X-KK - Sbdags modded _that11  Omni kernel"
echo "8  - ZOMBi-X-KK - hardslog Omni Grimlock"
echo "9  - ZOMBi-X-KK - Omni base kernel"
echo "Any other key exits"
read kernel_ver
echo "Mounting /system r/w..."
mount -o remount,rw -t ext4 /dev/block/mmcblk0p1 /system || echo "/system not mounted r/w"
mkdir -p /system/boot
case $kernel_ver in
    1) cp CROMi-X_that10/* /system/boot/ ;;
    2) cp CROMi-X_Stock/* /system/boot/ ;;
    3) cp omni-zombi-pop/* /system/boot/ ;;
    4) cp bootinit115-stock-zombie-pop/* /system/boot/ ;;
    5) cp grim300511-zombi-pop/* /system/boot/ ;;
    6) cp grimzombilite5-zombi-pop/* /system/boot/ ;;	
    7) cp that11cm11-300-omni-kk/* /system/boot/ ;;
    8) cp grimomni4-omni-kk/* /system/boot/ ;;
    9) cp bootinit102-300-omni-kk/* /system/boot/ ;;	
    *) mount -o remount,ro -t ext4 /dev/block/mmcblk0p1 /system || echo "/system not mounted r/o"
    exit 1
esac
echo "Remount /system r/o..." 
mount -o remount,ro -t ext4 /dev/block/mmcblk0p1 /system || echo "/system not mounted r/o"
echo ""
echo ""
echo "Android Kexec kernel installer - Done!"
echo ""
#rootfs install
#show busybox version
bb_ver=$(busybox|head -n1|cut -d " " -f2)
echo ""
echo "************************************"
echo "   Your Busybox version is:"
echo "   $bb_ver"
echo ""
echo "   You need at least 1.22.1"
echo "   to use this installer."
echo "   Press enter to continue, ctl+c to exit"
echo ""
echo "************************************"
echo ""
read
echo "************************************"
echo ""
use_lzma=$(ls ./*.lzma|wc -w)
if [ "$use_lzma" != "1" ] ; then
    echo "Exiting..."
    echo ""
    echo "Please make sure you place ONLY ONE"
    echo "rootfs.tar.lzma file in your install"
    echo "directory. Remove any others"
    echo "and You will now need to"
    echo "run the main installer again starting with menu item 4."
    echo ""
    echo "Enter to continue"
    read
    exit 1
fi
lzma_name=$(ls *.lzma)
echo ""
echo "   Using: $lzma_name for install"
echo ""
#set rootfs image name
echo ""
echo "************************************"
echo "Current rootfs image files (if any)"
echo ""
ls $kit
echo ""
echo "   Enter rootfs image name"
echo "   (ie my_lubuntu_may_14.img)"
echo "   Make sure this name is unique"
echo "   And include the .img at the end"
echo ""
echo "   The default boot.cfg entry is"
echo "   rootfs.img, use this if you do"
echo "   not want to modify the boot.cfg file."
echo ""
echo "************************************"
echo ""
echo ""
echo ""
read rootfs_name
#check for valid entry
if [ "$rootfs_name" == "" ] ; then
	echo "Invalid input. you will now need to"
    echo "run the main installer again starting with menu item 4."
	echo ""
	exit 1
fi
echo ""
#add extension if missed
correctext="$(echo "$rootfs_name" | rev | cut -d"." -f 1 | rev)"
if [ "$correctext" != "img" ] ; then
	rootfs_name="$(echo "$rootfs_name"| cut -d"." -f 1)".img
fi
#check if image name exists
if [ -f "$kit/$rootfs_name" ] ; then
    echo "File exists!"
    echo "Please run the main installer again, menu item 4"
    echo " and type a UNIQUE name carefully."
    echo ""
    echo "Here are the current image names:"
    echo ""
    ls $kit/
    echo "Enter to continue"
    read
    exit 1
fi
#check for internalsd space
inter_a=$(df /data | awk 'NR==2{print$4}'|tail -c 2)
inter_b=$(df /data | awk 'NR==2{print$4}')
inter_c=${inter_b%.*}
if [ ! $inter_a = "G" ] ; then
	echo "You have less than 1GB of free"
	echo "on your internal SD"
	echo "You will need to make room for"
	echo "the installer to work. Minimum is"
	echo "4GB. You only have $inter_b"
	echo ""
	echo "After clearing space, you will need to"
    echo "run the main installer again starting with menu item 4."
    echo ""
    echo "Press enter to continue..."
    echo ""
	read
	exit 1
fi
if [ "$inter_c" -lt "4" ] ; then
	echo "You have less than 4GB of free"
	echo "on your internal SD"
	echo "You will need to make room for"
	echo "the installer to work. Minimum is"
	echo "4GB. You only have $inter_b"
	echo ""
	echo "After clearing space, you will need to"
    echo "run the main installer again starting with menu item 4."
    echo ""
    echo "Press enter to continue..."
    echo ""
	read
	exit 1
fi
#select rootfs image size
#select rootfs image size
if grep -q crombi < /system/build.prop || grep -q zombi < /system/build.prop ; then
	rootfs_size=3
	else
	echo ""
	echo "************************************"
	echo ""
	echo "   Space avialable on Internal SD:"
	echo "   $inter_b"
	echo ""
	echo "   Select desired rootfs image size"
	echo ""
	echo "   1) 3 - GB"
	echo "   2) 4 - GB"
	echo "   3) 5 - GB"
	echo "   4) 10 - GB"
	echo "   5) 20 - GB"
	echo "   Any other key to exit"
	echo ""
	read z
	echo ""
	case $z in
		1) rootfs_size=3 ;;
		2) rootfs_size=4 ;;
		3) rootfs_size=5 ;;
		4) rootfs_size=10 ;;
		5) rootfs_size=20 ;;
		*) echo "Invalid selection, run installer again"
			echo ""
		exit
	esac
fi
#check rootfs image size againt available space
if [ "$rootfs_size" -gt "$inter_c" ] ; then
	echo "Not enough room on the internal sd."
	echo "Internal space= $inter_b , You chose $rootfs_size"
	echo "You will need to"
	echo "run the main installer again starting with menu item 4,"
	echo "and choose a smaller size for the image"
	echo ""
	echo "Press enter to continue"
	read
	exit 1
fi
echo ""
echo ""
echo "************************************"
echo ""
echo "   The installer will now create a new"
echo "   rootfs image file named:"
echo "   $rootfs_name"
echo "   in $kit"
echo "   Image size is: $rootfs_size GB"
echo ""
echo "   Press enter to continue, ctl+c to exit"
echo ""
echo "************************************"
read
START=$(date +%s)
echo ""
echo "   Creating Image flie..."
echo ""
echo "************************************"
echo ""
#mount system rw
mount -o remount,rw -t ext4 /dev/block/mmcblk0p1 /system
#create image file
busybox dd if=/dev/zero of="$kit"/"$rootfs_name" bs=1048576 seek="$rootfs_size"000 count=0 || echo "File created anyway"
echo ""
echo "Formating image file..."
echo ""
if grep -q crombi < /system/build.prop ; then
	mke2fs -F -t ext4 "$kit"/"$rootfs_name"
	else
	busybox mke2fs -F "$kit"/"$rootfs_name"
fi
#mount loop device
echo ""
echo "Mounting loop device..."
echo ""
if [ -b /dev/block/loop250 ]; then
	echo "Loop device exists"
else
busybox mknod /dev/block/loop250 b 7 250
fi
echo ""
busybox losetup /dev/block/loop250 "$kit"/"$rootfs_name"
busybox mount -t ext4 /dev/block/loop250 $mnt
echo "   Copying rootfs files to image"
echo "   ~10-30 minutes..."
cd $mnt
tar -ahxf "$installer_path"/"$lzma_name"
cd "$installer_path"
echo ""
echo "********************************"
echo ""
echo "   Rootfs copy done. "
echo ""
FINISH=$(date +%s)
echo "total rootfs install time: $(( ($FINISH-$START) / 60 )) minutes, $(( ($FINISH-$START) % 60 )) seconds"
echo ""
echo "   Unmounting image..."
sleep 15
busybox umount $mnt
busybox losetup -d /dev/block/loop250 || echo "Loop device not disassociated, Please reboot before using rootfs installer again."
mount -o remount,ro -t ext4 /dev/block/mmcblk0p1 /system
#add boot.cfg
mkdir -p /data/media/0/kexecbootcfg
mount -t vfat /dev/block/mmcblk0p5 /data/media/0/kexecbootcfg/
echo ""
echo " Installing boot.cfg..."
echo ""
mkdir -p /data/media/0/kexecbootcfg/multiboot/
echo '#Auto entry from rootfs installer script - 1st install#' > /data/media/0/kexecbootcfg/multiboot/boot.cfg
echo '#Android#' >> /data/media/0/kexecbootcfg/multiboot/boot.cfg
echo "LABEL=CROMi-X or CROMBi-KK" >> /data/media/0/kexecbootcfg/multiboot/boot.cfg
echo "BOOT=3" >> /data/media/0/kexecbootcfg/multiboot/boot.cfg
echo "DEVICE=/dev/mmcblk0p1" >> /data/media/0/kexecbootcfg/multiboot/boot.cfg
echo "DIR=/" >> /data/media/0/kexecbootcfg/multiboot/boot.cfg
echo "KERNEL=/boot/zImage" >> /data/media/0/kexecbootcfg/multiboot/boot.cfg
echo "INITRD=/boot/initrd.img" >> /data/media/0/kexecbootcfg/multiboot/boot.cfg
echo "PRIORITY=100" >> /data/media/0/kexecbootcfg/multiboot/boot.cfg
echo "" >> /data/media/0/kexecbootcfg/multiboot/boot.cfg
echo '#Auto entry from rootfs installer script#' >> /data/media/0/kexecbootcfg/multiboot/boot.cfg
echo "LABEL=$rootfs_name file on /data/media/linux" >> /data/media/0/kexecbootcfg/multiboot/boot.cfg
echo "BOOT=7" >> /data/media/0/kexecbootcfg/multiboot/boot.cfg
echo "DEVICE=/dev/mmcblk0p8" >> /data/media/0/kexecbootcfg/multiboot/boot.cfg
echo "IMAGE=/media/linux/$rootfs_name" >> /data/media/0/kexecbootcfg/multiboot/boot.cfg
echo "KERNEL=/boot/zImage" >> /data/media/0/kexecbootcfg/multiboot/boot.cfg
echo "INITRD=/boot/initrd.img" >> /data/media/0/kexecbootcfg/multiboot/boot.cfg
echo "PRIORITY=99" >> /data/media/0/kexecbootcfg/multiboot/boot.cfg
echo ""
echo ""
echo ""
echo ""
echo "Waiting for sync..."
sleep 10
umount /data/media/0/kexecbootcfg/
rm -r /data/media/0/kexecbootcfg/
echo ""
echo "   Boot.cfg installed..."
echo ""
echo "    Install complete."
echo ""
echo "   The kexecboot kernel, Android kexec kernel, a boot.cfg and"
echo "   a rootfs named : $kit/$rootfs_name"
echo "   have been installed."
echo ""
echo "   You can now reboot and profit!"


